src/components/library/CitationSearch.tsx

```diff
@@ -126,6 +128,26 @@ export const CitationSearch: React.FC<{
     [filterLibraryItems]
   )
 
+  const [
+    cancelTokenSource,
+    setCancelTokenSource,
+  ] = useState<CancelTokenSource>()
+
+  const apiSearchCallback = useCallback(
+    (
+      query: string,
+      params: { rows: number; sort?: string },
+      mailto: string
+    ) => {
+      cancelTokenSource?.cancel()
```

Comment:
What really happens here? Do previous requests slow down the server response somehow?

Reply:
Resolving to unblock merging.

Reply:
> What really happens here?

We need to cancel old requests, so when the user clicks to show more results it will be the only request to get references. and if we leave the old requests and they finish after showing more requests it's going to ruin the references list